File: C++/1051.cpp
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
public:
    int heightChecker(vector<int>& heights) {
      int res = 0;
        vector<int> s = heights;
        
        sort(s.begin() , s.end());
        
        for(int i = 0 ; i < s.size(); i++)
            res += heights[i] != s[i];
         return res;
    }    
};
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
    const static int RANGE = 100;
    int count[RANGE + 1];
    
    vector<int> csort(vector<int>& arr, int n) {
        memset(count, 0, sizeof(count));
        vector<int> ret(n);
        for (int t : arr)
            count[t]++;
        int j = 0;
        for (int i = 1; i <= RANGE; i++) {
            while (count[i]--)
                ret[j++] = i;
        }
        return ret;
    }
    
public:
    int heightChecker(vector<int>& heights) {
        int n = heights.size();
        vector<int> temp = csort(heights, n);
        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (temp[i] != heights[i])
                ans++;
        }
        return ans;
    }
};
__________________________________________________________________________________________________

File: Java/1051.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    public int heightChecker(int[] heights) {
        int[] count = new int[101] ;
        for(int i : heights) {
            count[i]++;
        }
        int[] lower = new int[101] ;
        int[] heigher = new int[101] ;
        int cnt = 0 ;
        for(int i = 1 ; i < 101 ; i++) {
            if(count[i] != 0) {
                lower[i] = cnt ;
                cnt += count[i] ;
                heigher[i] = cnt - 1;
            }
        }
        int res = 0 ;
        for(int i = 0 ; i < heights.length ; i++) {
            if((i )< lower[heights[i]] || (i) > heigher[heights[i]]) {
                res++;
            }
        }
        return res ;
    }
}
__________________________________________________________________________________________________
sample 1 ms submission
class Solution {
    public int heightChecker(int[] heights) {
        int[] dup = (int[])heights.clone();
        Arrays.sort(dup);
        int res = 0;
        for (int i = 0; i < heights.length; i++) {
            if (heights[i] != dup[i]) {
                res++;
            }
        }
        return res;       
    }
}
__________________________________________________________________________________________________
